core/src/main/java/org/ojai/util/MapEncoder.java

```diff
@@ -99,7 +99,7 @@
       switch (event) {
       case END_MAP:
       case END_ARRAY:
-        throw new EncodingException("Unexpected event: " + event);
+        return objList; // in case of empty array [] return the value immediately
```

Comment:
Should we continue to throw exception in case of END_MAP?

Reply:
yes that makes sense!